lib/galaxy/schema/invocation.py

```diff
@@ -218,3 +257,295 @@ class InvocationMessageResponseModel(BaseModel):
+    )
+    # TODO The state can differ from InvocationStepState is this intended?
+    # InvocationStepState is equal to the states attribute of the WorkflowInvocationStep class
+    state: Optional[ExtendedInvocationStepState] = Field(
+        default=None,
+        title="State of the invocation step",
+        description="Describes where in the scheduling process the workflow invocation step is.",
+    )
+    action: Optional[bool] = InvocationStepActionField
+    order_index: int = Field(
+        ...,
+        title="Order index",
+        description="The index of the workflow step in the workflow.",
+    )
+    workflow_step_label: Optional[str] = Field(
+        default=None,
+        title="Step label",
+        description="The label of the workflow step",
+    )
+    workflow_step_uuid: Optional[UUID4] = Field(
+        None,
+        title="UUID",
+        description="Universal unique identifier of the workflow step.",
+    )
+    outputs: Dict[str, InvocationStepOutput] = Field(
+        {},
+        title="Outputs",
+        description="The outputs of the workflow invocation step.",
+    )
+    output_collections: Dict[str, InvocationStepCollectionOutput] = Field(
+        {},
+        title="Output collections",
+        description="The dataset collection outputs of the workflow invocation step.",
+    )
+    jobs: List[schema.JobBaseModel] = Field(
+        [],
+        title="Jobs",
+        description="Jobs associated with the workflow invocation step.",
+    )
+
+
+class InvocationReport(Model):
+    """Report describing workflow invocation"""
+
+    render_format: REPORT_RENDER_FORMAT_MARKDOWN = Field(
+        literal_to_value(REPORT_RENDER_FORMAT_MARKDOWN),
+        title="Render format",
+        description="Format of the invocation report.",
+    )
+    markdown: Optional[str] = Field(
+        default="",
+        title="Markdown",
+        description="Raw galaxy-flavored markdown contents of the report.",
+    )
+    invocation_markdown: Optional[str] = Field(
+        default="",
+        title="Markdown",
+        description="Raw galaxy-flavored markdown contents of the report.",
+    )
+    model_class: schema.INVOCATION_REPORT_MODEL_CLASS = ModelClassField(schema.INVOCATION_REPORT_MODEL_CLASS)
+    id: EncodedDatabaseIdField = Field(
+        ...,
+        title="Workflow ID",
+        description="The workflow this invocation has been triggered for.",
+    )
+    username: str = Field(
+        ...,
+        title="Username",
+        description="The name of the user who owns this report.",
+    )
+    title: str = Field(
+        ...,
+        title="Title",
+        description="The name of the report.",
+    )
+    generate_time: Optional[str] = schema.GenerateTimeField
+    generate_version: Optional[str] = schema.GenerateVersionField
+
+    class Config:
+        # Galaxy Report/Page response can contain many extra_rendering_data
+        # Allow any other extra fields
+        extra = Extra.allow
```

Comment:
Only a fairly limited amount of extra keys are possible, `errors`, `generate_time`, `generate_version`, `history_datasets`, `workflows`, `history_dataset_collections`, `jobs`, `histories` and `invocations` afaict. We should just list them explicitly, otherwise we'll never catch misspelled attributes in the client.